src/app-logic/url-handling.js

```diff
@@ -42,27 +42,45 @@ export const CURRENT_URL_VERSION = 5;
  * to manage this bit of state.
  */
 let _isReplaceState: boolean = false;
+let _replaceHistoryCallCount: number = 0;
+
+function _enableHistoryReplaceState(): void {
+  _replaceHistoryCallCount++;
+  _isReplaceState = true;
+}
 
 /**
- * This function can be called from thunk actions or other components to change the
- * history API's behavior.
+ * Only disable the replace state if the call count is 0.
  */
-export function setHistoryReplaceState(value: boolean): void {
-  if (_isReplaceState === value) {
-    throw new Error(
-      `Trying to toggle the history replace state to: ${value.toString()} but it was already ${_isReplaceState.toString()}.`
-    );
+function _maybeDisableHistoryReplaceState(): void {
+  _replaceHistoryCallCount--;
+  if (_replaceHistoryCallCount < 0) {
+    throw new Error('_replaceHistoryCallCount cannot be negative.');
```

Comment:
Maybe the error could be: `_maybeDisableHistoryReplaceState was called more than _enableHistoryReplaceState which should never happen.`